### ISSUE TITLE
Error: ToSql not implemented

### ISSUE THREAD

**Issue by theduke** [adwhit/diesel-derive-enum#12]
Hey, I just tried out this crate.

Initial setup works fine, but I'm getting this error when inserting:

```rust
diesel::insert_into(db_schema::contacts::table).values(&contact).execute(&*db);
```

```
error[E0277]: the trait bound `db::types::Gender: diesel::serialize::ToSql<diesel::sql_types::Nullable<db::types::pg_enum_impl_Gender::GenderMapping>, diesel::pg::Pg>` is not satisfied
  --> src/app/users/mod.rs:55:78
   |
55 |             diesel::insert_into(db_schema::contacts::table).values(&contact).execute(&*db);
   |                                                                              ^^^^^^^ the trait `diesel::serialize::ToSql<diesel::sql_types::Nullable<db::types::pg_enum_impl_Gender::GenderMapping>, diesel::pg::Pg>` is not implemented for `db::types::Gender`
   |

```

Relevant types:

```rust

#[derive(Serialize, Deserialize, Identifiable, Queryable, Insertable, Clone, Debug)]
#[table_name = "contacts"]
pub struct Contact{
    pub id: Uuid,
    pub gender: Option<Gender>,
    pub given_name: Option<String>,
    pub family_name: Option<String>,
    pub company_name: Option<String>,
    pub title: Option<String>,
    pub phone_landline: Option<String>,
    pub phone_mobile: Option<String>,
    pub email: Option<String>,
    pub birthday: Option<NaiveDate>,
    pub address_id: Option<Uuid>,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

#[derive(PgEnum, Serialize, Deserialize, Clone, Copy, PartialEq, Eq, Debug)]
pub enum Gender {
    Male,
    Female,
    Other,
}

#[derive(Serialize, Deserialize, Identifiable, Queryable, Insertable, Clone, Debug)]
#[table_name = "contacts"]
pub struct Contact{
    pub id: Uuid,
    pub gender: Option<Gender>,
    pub given_name: Option<String>,
    pub family_name: Option<String>,
    pub company_name: Option<String>,
    pub title: Option<String>,
    pub phone_landline: Option<String>,
    pub phone_mobile: Option<String>,
    pub email: Option<String>,
    pub birthday: Option<NaiveDate>,
    pub address_id: Option<Uuid>,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

table! {
    use diesel::pg::types::sql_types::*;
    use diesel::sql_types::*;
    use db::types::GenderMapping;

    contacts (id) {
        id -> Uuid,
        gender -> Nullable<GenderMapping>,
        given_name -> Nullable<Text>,
        family_name -> Nullable<Text>,
        company_name -> Nullable<Text>,
        title -> Nullable<Text>,
        phone_landline -> Nullable<Text>,
        phone_mobile -> Nullable<Text>,
        email -> Nullable<Text>,
        birthday -> Nullable<Date>,
        address_id -> Nullable<Uuid>,
        created_at -> Timestamptz,
        updated_at -> Timestamptz,
    }
}
```

diesel: `1.1`
diesel-derive-enum: `0.3.0`

**Comment by adwhit** [comment]
Thanks for the bug report. Fix is in `0.4.1`

**Comment by theduke** [comment]
@adwhit I've tried with 0.4.1.

Still not working.

**Comment by adwhit** [comment]
Hmm have you set `features = ["postgres"]`?

**Comment by theduke** [comment]
Yep.

**Comment by adwhit** [comment]
Sorry, can't reproduce this. Chance you could post a snippet? (SQL and Rust)

**Comment by theduke** [comment]
Will do.

**Comment by berwyn** [comment]
I'm actually having a similar issue on queries, where multiple trait bounds fail:
`models::foo::Bar: diesel::Queryable<models::foo::db_enum_impl_Bar::BarMapping, diesel::pg::Pg>`
and
`models::foo::Bar: diesel::sql_types::HasSqlType<models::foo::db_enum_impl_Bar::BarMapping>`
in my case.

In case it helps, in `src/models/foo.rs`:
```rust
#[derive(DbEnum)]
pub enum Bar {
    A,
    B,
    C,
}

table! {
    use diesel::sql_types::*;
    use super::BarMapping;

    foos (id) {
        id -> Integer
        // *snip*
        bar -> BarMapping
    }
}

#[derive(Indentifiable, Queryable)]
#[table_name = "foos"]
pub struct Foo {
    // *snip*
    pub bar: Bar,
}
```

and in `migrations/blah/up.sql`:
```sql
CREATE TYPE bar AS ENUM (
    'a',
    'b',
    'c'
);

CREATE TABLE foos (
    -- *snip*
    bar bar NOT NULL
);
```

Lastly, the call that fails:
```rust
foos::table
    .limit(5)
    .load::<Foo>(&pg_connection)
    .expect("Boom 💥");
```

**Comment by adwhit** [comment]
@berwyn thanks, I'm looking into it

**Comment by lancecarlson** [comment]
I think I got burned by this exact issue as well. Can't get everything to completely compile given your exact instructions.

```
error[E0277]: the trait bound `diesel::pg::Pg: diesel::sql_types::HasSqlType<models::transaction_card::db_enum_impl_CardNetwork::Card_network>` is not satisfied
   |
38 |         .execute(&*conn)?;
   |          ^^^^^^^ the trait `diesel::sql_types::HasSqlType<models::transaction_card::db_enum_impl_CardNetwork::Card_network>` is not implemented for `diesel::pg::Pg`
```

**Comment by adwhit** [comment]
Thanks for bug reports. I have yanked 0.4.1, fixed in 0.4.2

**Comment by berwyn** [comment]
Confirmed, my case compiles now. Thanks for the quick turnaround!
